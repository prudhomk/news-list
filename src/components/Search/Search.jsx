import React from 'react';
import PropTypes from 'prop-types';

const Search = ({ search, onChange, onSubmit }) => {
  console.log(search);
  return (
    <form onSubmit={onSubmit}>
  
      <label htmlFor="search">Search Articles</label>
      <input
        id="search"
        type="search"
        name="search"
        value={search}
        onChange={onChange}
        placeholder="Search"
      />
    
      <button>Search</button>
    </form>
  );
};

Search.propTypes = {
  search: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default Search;

