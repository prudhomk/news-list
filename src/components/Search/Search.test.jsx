import React from 'react';
import { cleanup, render } from '@testing-library/react';
import Search from '../Search/Search';

describe('Search component', () => {
  afterEach(() => cleanup());
  it('renders search', async () => {
    const { asFragment } = render(<Search />);
    expect(asFragment()).toMatchSnapshot();
  });
});
