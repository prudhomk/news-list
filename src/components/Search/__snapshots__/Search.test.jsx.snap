// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders search 1`] = `
<DocumentFragment>
  <form>
    <label
      for="search"
    >
      Search Articles
    </label>
    <input
      id="search"
      name="search"
      placeholder="Search"
      type="search"
      value=""
    />
    <button>
      Search
    </button>
  </form>
</DocumentFragment>
`;
